### PR TITLE
Switch extenral-dns back to official upstream

### LINKED ISSUE #1744: Switch to upstream external-dns
* Currently, we maintain a fork of https://github.com/k8gb-io/external-dns. This imposes maintenance costs and security risks of stale dependencies within an image, as we rebuild it rarely. 
* The main goal of the fork was to unblock external-dns usage with NS record support.
* We need to evaluate the switch back to the original https://github.com/kubernetes-sigs/external-dns after the project evolved. 
* It looks like it supports NS names for Azure and other providers out of the box https://github.com/kubernetes-sigs/external-dns/blob/master/provider/azure/azure.go#L195 and https://github.com/kubernetes-sigs/external-dns/blob/master/provider/recordfilter.go#L21-L23 
```go
// SupportedRecordType returns true only for supported record types.
// Currently A, AAAA, CNAME, SRV, TXT and NS record types are supported.
func SupportedRecordType(recordType string) bool {
	switch recordType {
	case "A", "AAAA", "CNAME", "SRV", "TXT", "NS":
		return true
	default:
		return false
	}
}
```

### PR DESCRIPTION
**Pull request by ytsarev**
Fixes #1744

This one needs extensive testing with various edge DNS environments. 

The initial local test with rfc2136 is successful.

..

<details>
  <summary>HOW TO RUN CI</summary>
---

  By default, all the checks will be run automatically. Furthermore, when changing website-related stuff, the preview will be generated by the netlify bot.

  ### Heavy tests
  Add the [`heavy-tests`](/k8gb-io/k8gb/issues?q=is%3A*+label%3Aheavy-tests) label on this PR if you want full-blown tests that include more than 2-cluster scenarios.

  ### Debug tests
  If the test suite is failing for you, you may want to try triggering `Re-run all jobs` (top right) with [debug logging](https://docs.github.com/actions/monitoring-and-troubleshooting-workflows/enabling-debug-logging) enabled. It will also make the [print debug](/k8gb-io/k8gb/blob/master/.github/actions/print-debug/action.yaml) action more verbose.

</details>

### CODE REVIEW

File: chart/k8gb/values.yaml [hunk 1]
```diff
@@ -69,11 +69,7 @@ externaldns:
   # -- extra volume mounts
   extraVolumeMounts: []
   # -- external-dns image repo:tag
-  # It is important to use the image from k8gb external-dns fork to get the full
-  # functionality. See links below
-  # https://github.com/k8gb-io/external-dns
-  # https://github.com/k8gb-io/external-dns/pkgs/container/external-dns
-  image: ghcr.io/k8gb-io/external-dns:v0.13.4-azure-ns-multiarch
+  image: registry.k8s.io/external-dns/external-dns:v0.15.0
```
Comment (abaguas) [inline chart/k8gb/values.yaml:72]:
```suggestion
  image: registry.k8s.io/external-dns/external-dns:v0.15.1
```
As discussed in slack and the community meeting, version 0.15.1 contains our upstream changes

**Comment by ytsarev** [comment]
Azure test is green with `0.15.1` . Tested it using an approach from https://github.com/k8gb-io/k8gb/pull/1773 , just using [Crossplane](https://github.com/ytsarev/k8gb/tree/azure-crossplane/dns-providers/azure/crossplane) instead of Terraform

```
k get gslb -A --context k3d-test-gslb1
NAMESPACE    NAME                        STRATEGY   GEOTAG
test-azure   failover-playground-istio   failover   eu
```

```
k get gslb -A --context k3d-test-gslb2
NAMESPACE    NAME                        STRATEGY   GEOTAG
test-azure   failover-playground-istio   failover   us
```

```
az network dns record-set a list --resource-group k8gb-test  --zone-name "$EDGE_DNS_ZONE" --output table
TTL    Fqdn                            Name              ProvisioningState    ResourceGroup
-----  ------------------------------  ----------------  -------------------  ---------------
5      gslb-ns-eu-cloud.k8gb-test.io.  gslb-ns-eu-cloud  Succeeded            k8gb-test
5      gslb-ns-us-cloud.k8gb-test.io.  gslb-ns-us-cloud  Succeeded            k8gb-test
```

```
az network dns record-set ns list --resource-group k8gb-test  --zone-name "$EDGE_DNS_ZONE" --output table
TTL     Fqdn                 Name    ProvisioningState    ResourceGroup
------  -------------------  ------  -------------------  ---------------
172800  k8gb-test.io.        @       Succeeded            k8gb-test
5       cloud.k8gb-test.io.  cloud   Succeeded            k8gb-test
```

**Comment by ytsarev** [comment]
aws route53 is green. **NOTE** I have to slightly modify the chart templates for smoother operations, mainly setting default aws region to avoid zone list failure
```
aws route53 list-resource-record-sets --hosted-zone-id /hostedzone/Z1234 --output=yaml
ResourceRecordSets:
- Name: k8gb-test.io.
  ResourceRecords:
  - Value: ns-201.awsdns-25.com.
  - Value: ns-2014.awsdns-59.co.uk.
  - Value: ns-1087.awsdns-07.org.
  - Value: ns-771.awsdns-32.net.
  TTL: 172800
  Type: NS
- Name: k8gb-test.io.
  ResourceRecords:
  - Value: ns-201.awsdns-25.com. awsdns-hostmaster.amazon.com. 1 7200 900 1209600
      86400
  TTL: 900
  Type: SOA
- Name: cloud.k8gb-test.io.
  ResourceRecords:
  - Value: gslb-ns-eu-cloud.k8gb-test.io
  - Value: gslb-ns-us-cloud.k8gb-test.io
  TTL: 5
  Type: NS
- Name: gslb-ns-eu-cloud.k8gb-test.io.
  ResourceRecords:
  - Value: 172.21.0.6
  - Value: 172.21.0.7
  TTL: 5
  Type: A
- Name: gslb-ns-us-cloud.k8gb-test.io.
  ResourceRecords:
  - Value: 172.21.0.10
  - Value: 172.21.0.11
  TTL: 5
  Type: A
```
